src/trunk/libs/xml/0.10/sc3ml_0.10__quakeml_1.2-RT.xsl

```diff
@@ -384,7 +395,8 @@
                  depending on timeUsed, horizontalSlownessUsed and backazimuthUsed values -->
             <xsl:choose>
                 <xsl:when test="scs:weight">
-                    <xsl:if test="((not(scs:timeUsed)) or (scs:timeUsed='true') or (scs:timeUsed='1'))">
+                    <xsl:if test="((scs:timeUsed='true') or (scs:timeUsed='1'))
+                                  or (not(scs:timeUsed|scs:horizontalSlownessUsed|scs:backazimuthUsed))">
```

Comment:
Please explain why you think it is necessary to change the mapping. Without knowledge about usage of time for the weight calculation we assume that it was used for backward compatibility.

Reply:
Lets see with this file:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<seiscomp xmlns="http://geofon.gfz-potsdam.de/ns/seiscomp3-schema/0.10" version="0.10">
  <EventParameters publicID="smi:eu.emsc/unid">
    <origin publicID="smi:www.iris.edu/ws/event/query?originId=7680412">
      <time>
        <value>2011-03-11T05:46:24.120000Z</value>
      </time>
      <latitude>
        <value>38.297</value>
      </latitude>
      <longitude>
        <value>142.373</value>
      </longitude>
      <arrival>
        <pickID>smi:ch.ethz.sed/pick/117634</pickID>
        <phase>Pn</phase>
        <timeUsed>true</timeUsed>
        <weight>0.48</weight>
      </arrival>
      <arrival>
        <pickID>smi:geonet.org.nz/pick/4965459</pickID>
        <phase>P*</phase>
        <horizontalSlownessUsed>true</horizontalSlownessUsed>
        <weight>0.48</weight>
      </arrival>
      <arrival>
        <pickID>smi:geonet.org.nz/pick/4965461</pickID>
        <phase>P*</phase>
        <timeUsed>true</timeUsed>
        <horizontalSlownessUsed>true</horizontalSlownessUsed>
        <backazimuthUsed>true</backazimuthUsed>
        <weight>0.48</weight>
      </arrival>
      <arrival>
        <pickID>smi:geonet.org.nz/pick/4965462</pickID>
        <phase>P*</phase>
        <weight>0.48</weight>
      </arrival>
    </origin>
    <event publicID="smi:ch.ethz.sed/event/historical/1165">
      <originReference>smi:www.iris.edu/ws/event/query?originId=7680412</originReference>
    </event>
  </EventParameters>
</seiscomp>
```

With your XSLT file, the arrival of the pick `smi:geonet.org.nz/pick/4965459` will be:

```xml
        <arrival publicID="smi:geonet.org.nz/pick/4965459_smi_www.iris.edu/ws/event/query?originId=7680412">
          <timeWeight>0.48</timeWeight>
          <horizontalSlownessWeight>0.48</horizontalSlownessWeight>
          <pickID>smi:geonet.org.nz/pick/4965459</pickID>
          <phase>P*</phase>
        </arrival>
```

Both `timeWeight` and `horizontalSlownessWeight` have been added while there was only an `horizontalSlownessUsed` node.

The change I made just adds a `timeWeight` node if none of the nodes `timeUsed`, `horizontalSlownessUsed` and `backazimuthUsed` have been defined. The arrival of the pick `smi:geonet.org.nz/pick/4965459` looks like:

```xml
        <arrival publicID="smi:geonet.org.nz/pick/4965459_smi_www.iris.edu/ws/event/query?originId=7680412">
          <horizontalSlownessWeight>0.48</horizontalSlownessWeight>
          <pickID>smi:geonet.org.nz/pick/4965459</pickID>
          <phase>P*</phase>
        </arrival>
```

However, the arrival `smi:geonet.org.nz/pick/4965462` still are in both case:

```xml
        <arrival publicID="smi:geonet.org.nz/pick/4965461_smi_www.iris.edu/ws/event/query?originId=7680412">
          <timeWeight>0.48</timeWeight>
          <pickID>smi:geonet.org.nz/pick/4965461</pickID>
          <phase>P*</phase>
        </arrival>
```